internal/webhooks/cluster.go

```diff
@@ -419,13 +421,20 @@ func validateMachineHealthChecks(cluster *clusterv1.Cluster, clusterClass *clust
 		// (One of these definitions will be used in the controller to create the MachineHealthCheck)
 
 		// Check if the machineHealthCheck is explicitly enabled in the ControlPlaneTopology.
-		if cluster.Spec.Topology.ControlPlane.MachineHealthCheck.Enable != nil && *cluster.Spec.Topology.ControlPlane.MachineHealthCheck.Enable {
-			// Ensure the MHC is defined in at least one of the ControlPlaneTopology of the Cluster or the ControlPlaneClass of the ClusterClass.
-			if cluster.Spec.Topology.ControlPlane.MachineHealthCheck.MachineHealthCheckClass.IsZero() && clusterClass.Spec.ControlPlane.MachineHealthCheck == nil {
-				allErrs = append(allErrs, field.Forbidden(
-					fldPath.Child("enable"),
-					fmt.Sprintf("cannot be set to %t as MachineHealthCheck definition is not available in the Cluster topology or the ClusterClass", *cluster.Spec.Topology.ControlPlane.MachineHealthCheck.Enable),
-				))
+		if cluster.Spec.Topology.ControlPlane.MachineHealthCheck.Enable != nil {
+			if *cluster.Spec.Topology.ControlPlane.MachineHealthCheck.Enable {
+				// Ensure the MHC is defined in at least one of the ControlPlaneTopology of the Cluster or the ControlPlaneClass of the ClusterClass.
+				if cluster.Spec.Topology.ControlPlane.MachineHealthCheck.MachineHealthCheckClass.IsZero() && clusterClass.Spec.ControlPlane.MachineHealthCheck == nil {
+					allErrs = append(allErrs, field.Forbidden(
+						fldPath.Child("enable"),
+						fmt.Sprintf("cannot be set to %t as MachineHealthCheck definition is not available in the Cluster topology or the ClusterClass", *cluster.Spec.Topology.ControlPlane.MachineHealthCheck.Enable),
+					))
+				}
+			} else {
+				// If enabled is set to false, the MachineHealthCheckClass field should not be set.
+				if !cluster.Spec.Topology.ControlPlane.MachineHealthCheck.MachineHealthCheckClass.IsZero() {
+					allWarnings = append(allWarnings, "if the MachineHealthCheck enabled is set to false, the MachineHealthCheckClass field should not be set.")
```

Comment:
```suggestion
					allWarnings = append(allWarnings, "MachineHealthCheck.Enable is set to false, class options won't be set")
```

or something similar

Reply:
Updated.